registrations/serializers.py

```diff
@@ -274,14 +273,131 @@ def validate(self, data):
 
 class SeatReservationCodeSerializer(serializers.ModelSerializer):
     timestamp = DateTimeField(default_timezone=pytz.UTC, required=False)
+
     expiration = serializers.SerializerMethodField()
 
-    class Meta:
-        fields = ("seats", "code", "timestamp", "registration", "expiration")
-        model = SeatReservationCode
+    in_waitlist = serializers.SerializerMethodField()
 
     def get_expiration(self, obj):
-        return obj.timestamp + timedelta(minutes=code_validity_duration(obj.seats))
+        return obj.expiration
+
+    def get_in_waitlist(self, obj):
+        registration = obj.registration
+        maximum_attendee_capacity = registration.maximum_attendee_capacity
+
+        if maximum_attendee_capacity is not None:
+            attendee_count = registration.signups.filter(
+                attendee_status=SignUp.AttendeeStatus.ATTENDING
+            ).count()
+
+            if maximum_attendee_capacity - attendee_count <= 0:
+                return True
+
+        return False
+
+    def validate(self, data):
+        instance = self.instance
+        errors = {}
+
+        # Amount of seats is always required
+        if data.get("seats") is None:
+            errors["seats"] = ErrorDetail(
+                _("This field must be specified."), code="required"
+            )
+
+        if isinstance(instance, SeatReservationCode):
+            non_editable_fields = ["registration", "timestamp"]
+
+            # It's not allowed to change registration or timestamp values
+            for field in non_editable_fields:
+                if field in data and getattr(instance, field) != data[field]:
+                    errors[field] = _("You may not change the value.")
```

Comment:
While you can make `timestamp` read-only, `registration` seems to be something that needs to be "create-only", no? Instead of this, you could override `to_internal_value` something like this:
```python
    def to_internal_value(self, data):
        if self.instance:
            data["registration"] = self.instance.registration.id
        return super().to_internal_value(data)
```
This way, it doesn't really matter what the value of the field is or if it even exists during update, it always gets replaced by the reservation's registration.